### PR TITLE
Fully qualifies images names for Istio samples.

### PR DESCRIPTION
**Pull request by ibm-jason**
Fixes: #14237.

Signed-off-by: Jason Clark <jason.clark@ibm.com>

This PR provides fully qualified `image:` fields for ~the bookinfo~ all Istio samples.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[X] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/grafana/values.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 enabled: false
 replicaCount: 1
 image:
-  repository: grafana/grafana
+  repository: docker.io/grafana/grafana
```
Comment (sdake) [inline install/kubernetes/helm/istio/charts/grafana/values.yaml]: grafana/grafana is not a hub/image combo - its some weird hardcode state of these values.  I'd recommend changing repository to hub, and permitting the new `hub` to be set to `docker.io/grafana` and adding an image name for the image name.  (image.name).  Third party repositories that Istio doesn't manage need to be able to specify both image and hub.

Cheers
-steve

Comment (ibm-jason) [reply install/kubernetes/helm/istio/charts/grafana/values.yaml]: @sdake thanks for the feedback.

This goes back to the way I modified it this recently closed PR:
https://github.com/istio/istio/pull/14312

In this PR, @linsun gave us the following feedback:
> Yes, for this PR, I would prefer to keep the old helm values API and add fully qualified image name. Then if you need this change in the new installer, you need a different PR for the installer repo.

I can modify this PR adhere to either approach, I think we just need to agree on which one to use. :-)

**Comment by ibm-jason** [comment]
> Why only bookinfo? For consistency, shouldn't this be done for all the samples?

@frankbu True. However, this PR was to directly address bookinfo as mentioned in #14237. If you want, I can create another issue to do this for all of the samples.

**Pull request merged**